packages/ui/src/components/ValidatorList/KeystoresDataGrid.tsx

```diff
@@ -61,6 +69,106 @@ export default function KeystoresDataGrid({
     openDashboardTab();
   }, [validatorSummaryURL]);
 
+  useState<CustomValidatorUpdateRequest>();
+  const [validatorsSubscriptionMap, setValidatorsData] =
+    useState<SmoothStatusByPubkey>({});
+  const [oracleCallError, setOracleCallError] = useState<string>();
+
+  // Check that Smooth API returns an expected response format
+  function isValidOracleResponse(response: SmoothValidatorByIndexApiResponse): boolean {
+    return (
```

Comment:
I might be out of context, why are we expecting those values as null? could you add a comment pls?

Reply:
We expect them to either exist or to be null.

The oracle call recieves the validator's indeces loaded in brain and returns two arrays, `found_validators` and `not_found_validators`. A validator that never interacted with oracle will be inside `not_found_validator`. A validator that at some point has already interacted with the oracle will be inside `found_validators`

For example, if a user loads web3signer brain for the first time and has never subscribed to smooth, we expect `found_validators` to be `null`, since its validators have not interactuated with the oracle yet and it is not tracking them.